src/System.ComponentModel.Annotations/tests/CompareAttributeTests.cs

```diff
@@ -6,10 +6,25 @@
 using System.Reflection;
 using Xunit;
 
-namespace System.ComponentModel.DataAnnotations
+namespace System.ComponentModel.DataAnnotations.Tests
 {
-    public class CompareAttributeTests
+    public class CompareAttributeTests : ValidationAttributeTestBase
     {
+        public override IEnumerable<Test> ValidValues() => new Test[]
+        {
+            new Test(new CompareAttribute("CompareProperty"), "test", new ValidationContext(new CompareObject("test"))),
+            new Test(new DerivedCompareAttribute("CompareProperty"), "a", new ValidationContext(new CompareObject("b")))
+        };
+
+        private static ValidationContext s_context = new ValidationContext(new CompareObject("a")) { DisplayName = "CurrentProperty" };
+        public override IEnumerable<Test> InvalidValues() => new Test[]
+        {
+            new Test(new CompareAttribute(nameof(CompareObject.CompareProperty)), "b", s_context),
+            new Test(new CompareAttribute(nameof(CompareObject.ComparePropertyWithDisplayName)), "b", s_context),
+            new Test(new CompareAttribute("NoSuchProperty"), "b", s_context),
+            new Test(new CompareAttribute(nameof(CompareObject.CompareProperty)), "b", new ValidationContext(new CompareObjectSubClass("a")))
+        };
```

Comment:
I'm wondering if this is really the best transformation.  This means that we've grouped all inputs into a single logical test, which means that if one fails, it prevents execution of subsequent tests, hiding potential failures.

Did you consider just having the base class provide a protected Validate method?  Then the validation logic is shared in a base class across all of the attribute tests, but the derived classes can still use InlineData and the like for the individual test cases, on a method that just calls to the base Validate method.